misk/src/main/kotlin/misk/web/metadata/DashboardMetadataAction.kt

```diff
@@ -75,7 +79,8 @@ class DashboardMetadataAction @Inject constructor() : WebAction {
     val home_url: String = "",
     val navbar_items: List<String> = listOf(),
     val navbar_status: String = "",
-    val tabs: List<DashboardTab> = listOf()
+    val tabs: List<DashboardTab> = listOf(),
+    val admin_dashboard: AdminDashboardConfig = AdminDashboardConfig()
```

Comment:
Should this match https://github.com/cashapp/misk-web/pull/1442/files#diff-c27559eac873e2f6fac51c844fe9fbbce4ead534bc37584157fba842244c9f8dR29 (i.e. `admin_dashboard_config`)?

Also, it shouldn't really cause problems in this case, but I'm a little reluctant to couple the config object and this action directly because it makes it easy to inadvertently output all parts of this config object (though really, I'm mostly just thinking about separation of concerns).

Reply:
> Should this match https://github.com/cashapp/misk-web/pull/1442/files#diff-c27559eac873e2f6fac51c844fe9fbbce4ead534bc37584157fba842244c9f8dR29 (i.e. admin_dashboard_config)?

Yep, thanks @r3mariano - I'll change misk-web to match.

> Also, it shouldn't really cause problems in this case, but I'm a little reluctant to couple the config object and this action directly because it makes it easy to inadvertently output all parts of this config object (though really, I'm mostly just thinking about separation of concerns).

What would be the alternative? Introduce a new endpoint to return this data and make another API call?

Reply:
Could introduce an object that can be created from the config, e.g. 
```kotlin
data class AdminDashboardConfigResponse(...) { 
  constructor(config: Config) {
    // ... 
  }
}
```

To get to that object, a pattern I usually see is to have the contents of the config provided at the module (this also makes testing easier at the module level). Though, you can still inject WebConfig directly, I'm not too fussed about it either way.

That reminds me, you should probably add a test that this field is outputted.

Reply:
> What would be the alternative? Introduce a new endpoint to return this data and make another API call?

Thinking this through, if we went down this path it could really simplify the solution. There'd be no need to modify misk-web at all if we introduced an additional API call from the web-actions `TabContainer.tsx`. WDYT?

Reply:
I don't follow, but if it makes it simpler that sounds good!

Reply:
Well, the approach in this PR is to extend the DashboardMetadata to include configuration for the admin dashboard. Because the call to retrieve the DashboardMetadata is made from within `misk-web` it results in us needing to use a react context provider there to provide the context to tabs rendered by misk. To make this simpler we could make an additional API call [when we mount the TabContainer](https://github.com/cashapp/misk/blob/7feb5edc7a55e72a1314070cfbae44561b9a8df7/misk/web/tabs/web-actions/src/rewrite/TabContainer.tsx#L14-L36) to retrieve the admin config.

Reply:
Do you need this data for all tabs or just web actions? If the latter, you might get an even simpler solution adding it to `WebActionMetadataAction`?

Reply:
Just for web actions. It's not something that will change per action though, just a single config value that the client can then use to construct a URL.

Reply:
Yes, then the response object can contain a new field that's not the list of actions

Reply:
Ok, I think I'll close this PR and start again. Thanks Ric 😄

Reply:
Agreed, `WebActionMetadataAction` is a more fitting spot to put this, good catch @r3mariano